### ISSUE TITLE
[Bug]: After 6.0.2, timeout no longer works

### ISSUE THREAD

**Issue by squintfox** [lextudio/pysnmp#50]
### Expected behavior

I expected this, UdpTransportTarget on getCmd() should time out when not able to connect to a target.  It does in <= 6.0.1.

### Actual behavior

When a timeout should occur, say a device isn't on the network, pysnmp hangs forever with no output.  It doesn't matter if you set timeout/retries or leave it as default.

### Detailed steps

```
import asyncio
from pysnmp.hlapi.asyncio import *
from pysnmp.smi.rfc1902 import ObjectIdentity, ObjectType

loop = asyncio.get_event_loop()
snmpEngine = SnmpEngine()


async def run_get():
    errorIndication, errorStatus, errorIndex, varBinds = await getCmd(
        snmpEngine,
        CommunityData('community_string'),
        UdpTransportTarget(('1.2.3.4', 161), timeout=2, retries=2),
        ContextData(),
        ObjectType(ObjectIdentity('1.3.6.1.2.1.1.1.0')),
    )
    for varBind in varBinds:
        print([str(varBind[0]), varBind[1]])


loop.run_until_complete(run_get())
snmpEngine.transportDispatcher.closeDispatcher()
```

### Python package information

6.0.2

### Operating system information

Windows 11

### Python information

3.11.7

**Comment by lextm** [comment]
Just shipped 6.0.4 release. Please verify if the issue has been resolved.

**Comment by squintfox** [comment]
Looks good to me.  Thanks @lextm.
